lib/resty/acme/storage/redis.lua

```diff
@@ -81,35 +81,32 @@ local function remove_namespace(namespace, keys)
   end
 end
 
--- TODO: use EX/NX flag if we can determine redis version (>=2.6.12)
 function _M:add(k, v, ttl)
   k = self.namespace .. k
-  local ok, err = op(self, 'setnx', k, v)
+  local ok, err
+  if ttl then
+    ok, err = op(self, 'set', k, v, "nx", "px", math.floor(ttl * 1000))
+  else
+    ok, err = op(self, 'set', k, v, "nx")
+  end
   if err then
     return err
-  elseif ok == 0 then
+  elseif ok == ngx.null then
```

Comment:
What happened here? I can't find `ngx.null` in the [docs](https://openresty-reference.readthedocs.io/en/latest/Lua_Nginx_API/?q=ngx.null&check_keywords=yes&area=default)

Reply:
I used `nil` at first, but this didn't work. I've noticed that other functions in this module use `ngx.null` instead.

Reply:
https://github.com/openresty/lua-resty-redis/issues/90